h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -494,11 +494,12 @@ private static long convertToMillis(Calendar cal, int year, int month, int day,
      * Get the specified field of a date, however with years normalized to
      * positive or negative, and month starting with 1.
      *
-     * @param d the date
+     * @param date the date value
      * @param field the field type
      * @return the value
      */
-    public static int getDatePart(java.util.Date d, int field) {
+    public static int getDatePart(Value date, int field) {
+        java.util.Date d = date.getTimestamp();
```

Comment:
getTimestamp->getDate

Reply:
This call was replaced by following commits with `valueToCalendar()` that calls proper methods for date-time values to avoid conversions between Value types and to provide support for `ValueTimestampTimezone`.